src/git-repository-provider.coffee

```diff
@@ -26,9 +26,9 @@ isValidGitDirectorySync = (directory) ->
   # To decide whether a directory has a valid .git folder, we use
   # the heuristic adopted by the valid_repository_path() function defined in
   # node_modules/git-utils/deps/libgit2/src/repository.c.
-  return directory.getSubdirectory('objects').existsSync() and
-      directory.getFile('HEAD').existsSync() and
-      directory.getSubdirectory('refs').existsSync()
+  return directory.getSubdirectory('objects').existsSync?() and
```

Comment:
I would say leave `isValidGitDirectorySync` as it was; we only need to check once if the `Directory` (and the directories it returns) implements synchronous IO; no need to check each call.